src/device/prims_simple.h

```diff
@@ -680,15 +698,11 @@ class Primitives<
 
     // if (barrierAny(flags & NetDeviceUnpack)) {
     //   flags |= AnyNetDeviceUnpack;
-    //   // g == 0 is the first ThreadPerSync # of threads of this warp
-    //   // g == 0 is also the RoleWaitRecv threads of this group, thus the thread ID will correlate to the peer index
-    //   if (g == 0) {
-    //     uint32_t mask = __ballot_sync((1U << ThreadPerSync) - 1, (flags & NetDeviceUnpack) ? 1 : 0);
-
-    //     // We only want to update the shared memory variable with a single thread
-    //     if (tid == 0) {
-    //       ncclShmem.groups[this->group].devicePlugin.unpack.unpackNetDeviceIndexMask = mask;
-    //     }
+    //   // RoleWaitRecv starts at tid=0, so this creates the bitmask of which recv peers
+    //   // have NetDeviceUnpack.
+    //   uint32_t mask = __ballot_sync(~0u, ((flags & RoleWaitRecv) && (flags & NetDeviceUnpack)) ? 1 : 0);
+    //   if (tid == 0) {
+    //     ncclShmem.groups[this->group].devicePlugin.unpack.unpackNetDeviceIndexMask = mask;
     //   }
     // }
```

Comment:
This code is not commented out in NCCL, but has been commented out in RCCL, so I've commented the changed code out in this merge.

Reply:
It is fine to leave it as is if it is currently not supported